file_transfer_protocol/server.py

```diff
@@ -3,32 +3,32 @@
 ONE_CONNECTION_ONLY = True      # Set this to False if you wish to continuously accept connections
 
 port = 12312                    # Reserve a port for your service.
-s = socket.socket()             # Create a socket object
+sock = socket.socket()             # Create a socket object
 host = socket.gethostname()     # Get local machine name
-s.bind((host, port))            # Bind to the port
-s.listen(5)                     # Now wait for client connection.
+sock.bind((host, port))            # Bind to the port
+sock.listen(5)                     # Now wait for client connection.
 
 print('Server listening....')
 
 while True:
-    conn, addr = s.accept()     # Establish connection with client.
+    conn, addr = sock.accept()     # Establish connection with client.
     print('Got connection from', addr)
     data = conn.recv(1024)
     print('Server received', repr(data))
 
     filename='mytext.txt'
-    f = open(filename,'rb')
-    l = f.read(1024)
-    while (l):
-       conn.send(l)
-       print('Sent ',repr(l))
-       l = f.read(1024)
-    f.close()
+    file = open(filename,'rb')
```

Comment:
Please change __file__ to __in_file__ to make the filemode clear to the reader.

Reply:
Use the __with open() as__ syntax to ensure that the file.close() happens even if an exception is thrown.